crates/storage/src/managers/chainstate.rs

```diff
@@ -5,19 +5,26 @@ use std::sync::Arc;
 use strata_db::{traits::*, DbResult};
 use strata_state::{chain_state::Chainstate, state_op::WriteBatch};
 use threadpool::ThreadPool;
+use tokio::sync::broadcast;
 
 use crate::{cache, ops};
 
 pub struct ChainstateManager {
     ops: ops::chainstate::ChainstateOps,
     wb_cache: cache::CacheTable<u64, Option<WriteBatch>>,
+    bcast_tx: broadcast::Sender<u64>,
 }
 
 impl ChainstateManager {
     pub fn new<D: Database + Sync + Send + 'static>(pool: ThreadPool, db: Arc<D>) -> Self {
         let ops = ops::chainstate::Context::new(db.chain_state_db().clone()).into_ops(pool);
         let wb_cache = cache::CacheTable::new(64.try_into().unwrap());
-        Self { ops, wb_cache }
+        let (bcast_tx, _) = broadcast::channel(1024);
+        Self {
+            ops,
+            wb_cache,
+            bcast_tx,
+        }
```

Comment:
This isn't really *not* the place for this.  We should really be using a higher level construct for this.  This means we're using something fairly low in the stack to signal logic much higher in the stack.

Reply:
I don't think this is true. The hierarchy is  "Database" > "Ops" > "Manager", ignoring rocksdb and rockbound.

Everything higher on the stack interacts with the `Manager`. So, I too think that the manager is very much qualified to notify the interacting components about the database changes, that also whenever one subscribes.

Reply:
It's a layering violation.  The source of truth that writes new chainstates (the FCM) is what those systems should really be listening to.  Components generally shouldn't be sending signals *up* the stack, especially when they go into different verticals.